flake-lang/flake-rust.nix

```
@@ -1,33 +1,68 @@
 pkgs:
 
-{ crane, src, crateName, extraSources ? [ ], extraSourcesDir ? ".extras", data ? [ ], dataDir ? "data", devShellHook ? "", devShellTools ? [ ], testTools ? [ ] }:
+{ crane
+, src
+, crateName
+, rustVersion ? "latest"
+, nativeBuildInputs ? [ ]
+, extraSources ? [ ]
+, extraSourcesDir ? ".extras"
+, data ? [ ]
+, dataDir ? "data"
+, devShellHook ? ""
+, devShellTools ? [ ]
+, testTools ? [ ]
+}:
 let
-  rustWithTools = pkgs.rust-bin.stable.latest.default.override {
+  rustWithTools = pkgs.rust-bin.stable.${rustVersion}.default.override {
     extensions = [ "rustfmt" "rust-analyzer" "clippy" "rust-src" ];
   };
   craneLib = crane.lib.${pkgs.system}.overrideToolchain rustWithTools;
 
-  # Library source code with extra dependencies attached
-  fullSrc = pkgs.stdenv.mkDerivation {
-    src = craneLib.cleanCargoSource (craneLib.path src);
-    name = "lbf-rust-build-env";
-    unpackPhase = ''
-      mkdir $out
-      cp -r $src/* $out
-      cd $out
-      ${copyExtraSources}
-      ${copyData}
-    '';
-  };
+  cleanSrc = craneLib.cleanCargoSource (craneLib.path src);
+
+  # Library source code with extra dependencies copied
+  buildEnv =
+    pkgs.stdenv.mkDerivation
+      {
+        src = cleanSrc;
+        name = "${crateName}-build-env";
+        unpackPhase = ''
+          mkdir $out
+          cp -r $src/* $out
+          cd $out
+          ${copyExtraSources}
+          ${copyData}
+        '';
+      };
+
+  # Library source code, intended to be in extra-sources (inside the `.extras` directory)
+  # The main difference is that dependencies are not copied, to `.extras`
+  # but they are referenced from the parent directory (`../`).
+  vendoredSrc =
+    pkgs.stdenv.mkDerivation
+      {
+        src = cleanSrc;
+        name = "${crateName}-vendored-src";
+        unpackPhase = ''
+          mkdir $out
+          cp -r $src/* $out
+          cd $out
+          sed -i 's/.extras/../g' Cargo.toml
```

```suggestion
          sed -i 's/\.extras/../g' Cargo.toml
```
I think we meant to escape the `.` here as `.` normally matches any singular character (sometimes it doesn't match new line, I may be misrecalling).

A minimal example to play with
```
$ echo ".extras" | sed 's/.extras/DOGS/g'
DOGS
$ echo "~extras" | sed 's/.extras/DOGS/g'  # This still matches, not sure if this is what we want
DOGS
$ echo ".extras" | sed 's/\.extras/DOGS/g'
DOCS
$ echo "~extras" | sed 's/\.extras/DOGS/g' # This doesn't match -- pretty sure we want this.
~extras
```

---

good catch, thanks!